modules/modelSetup/BaseModelSetup.py

```diff
@@ -73,15 +75,20 @@ def predict(
     ) -> dict:
         pass
 
-    @abstractmethod
```

Comment:
Hey, is there a reason you included this refactoring in this PR? I think the ModelSetupDiffusionLossMixin being a mixin usable by the subclasses of this, rather than included in the base class, is something of a Chesterton's Fence.

Reply:
Yeah -- I noticed that ALL models except Würstchen had exactly the same variant of this function, so I moved it to the base model and let Würstchen be as-is instead. It's reducing code repetition.

Reply:
The reason for this separation was that not all models need to be diffusion models. It's not really relevant right now, since no other model type is supported right now, but it might be relevant in the future. I just wanted to keep this clear in the code, so decided to include the function call in all sub classes

Reply:
I'll move the inheritance to the subclasses, but keep the code in the parent class to reduce most of the repetition.

Reply:
I don't think this is resolved. The parent class is still not necessarily representing a diffusion model, so the calling of the _diffusion_losses method (which is defined in the diffusion mixin) shouldn't leak into this abstract parent. If you wanted to factor it out, you could in this file create a subclass of this abstract parent that's just for diffusion models, implements the mixin, and put the implementation of calculate_loss that calls the _diffusion_losses method there.

Reply:
https://github.com/Nerogar/OneTrainer/pull/212/commits/14f2a3238f5faacfb65c8cfb4eca68387f44565b

Reply:
Went back to the old way of doing things in https://github.com/Nerogar/OneTrainer/pull/212/commits/94506c6692da3eabdbba082e902b36747192feec.